eth/vm/forks/berlin/__init__.py

```diff
@@ -52,14 +41,15 @@ class BerlinVM(MuirGlacierVM):
     compute_difficulty = staticmethod(compute_berlin_difficulty)    # type: ignore
     configure_header = configure_berlin_header
 
-    @staticmethod
+    @classmethod
     def make_receipt(
+            cls,
             base_header: BlockHeaderAPI,
             transaction: SignedTransactionAPI,
             computation: ComputationAPI,
             state: StateAPI) -> ReceiptAPI:
 
-        gas_used = base_header.gas_used + finalize_gas_used(transaction, computation)
+        gas_used = base_header.gas_used + cls.finalize_gas_used(transaction, computation)
```

Comment:
If you think it's a good idea (I'm on the fence), it should be possible to backport this berlin implementation of `make_receipt()` to byzantium. Then this wouldn't be a custom method at all here, AFAICT.

Reply:
Hmm, I don't have a good feel for why this custom method was added in the first place. Do you know the background?

Reply:
It's just because Berlin now has different transaction types, which define different receipt types. So it has to ask the transaction to make the receipt. But that should continue to work in pre-Berlin, I think. It just never got backported. It's fine to skip this, if that doesn't make sense. The situation is certainly not any worse than it was before.

Reply:
Got it. That makes sense. Yeah, I think I like it slightly better to always ask the transaction 👍 I will update!